internal/client/config.go

```diff
@@ -56,6 +70,74 @@ type AuthInfo struct {
 	ClientKeyData []byte `json:"client-key-data,omitempty" datapolicy:"security-key"`
 }
 
+func (c *Config) Equal(c2 *Config) bool {
+	if c == c2 {
+		return true
+	}
+	if c == nil || c2 == nil {
```

Comment:
the check above would tolerate both objects being nil. Config should always be created with a constructor. Trying to understand what case c would be nil I guess if I did something like below which would deserve a panic.
```
var c *Config
```

Reply:
If both are `nil`, they are Equal() by definition. I also think the receiver should never be `nil`, but if someone used `cmd.Equal()` on Configs, then this seems to be a case that can happen (https://pkg.go.dev/github.com/google/go-cmp/cmp#Equal).

Reply:
I still don't understand how c could be nil but this is non-blocking